src/libs/datatableproviders/DataProvider.ts

```diff
@@ -0,0 +1,78 @@
+// define metadata structures
+interface EntityTypeMetadata {
+  attributeNames: string[],
+  count: number,
+  idName: string
+}
+
+export interface EntityMetadata {
+  [index: string]: EntityTypeMetadata
+}
+
+export enum EntityQuerySortDirection {
+  // eslint-disable-next-line no-unused-vars
+  Asc = 'asc',
+  // eslint-disable-next-line no-unused-vars
+  Desc = 'desc'
+}
+
+export enum EntityQueryFilterOperator {
+  // eslint-disable-next-line no-unused-vars
+  And = 'and',
+  // eslint-disable-next-line no-unused-vars
+  Or = 'or'
+}
+
+// define paginated query result structures
+interface EntityQuery {
+  page: number,
+  pageSize: number,
+  sortField: string,
+  sortDirection: EntityQuerySortDirection,
+  filterTerms: string,
+  filterOperator: EntityQueryFilterOperator
+}
+interface EntityQueryResultMetadata {
+  unfilteredCount: number,
+  filteredCount: number,
+  filteredPageCount: number
+}
+interface Entity {
+  name: string,
+  entityType: string,
+  attributes: Record<string, any>
+}
+
+export interface EntityQueryResponse {
+  parameters: EntityQuery,
+  resultMetadata: EntityQueryResultMetadata
```

Comment:
Members in other interfaces in this file are delimited with a comma. Commas aren't necessary, but it would be nice to establish a convention early (possibly enforced with ESLint's [member-delimiter-style](https://typescript-eslint.io/rules/member-delimiter-style/)).

Reply:
Not to turn this into a bike shed, but I'd vote for no commas. I think not requiring unnecessary syntax is more in line with other Terra UI style conventions (in particular, the convention that we don't use semicolons).